### DISCUSSION TITLE
Improve the shadow DOM support

### DISCUSSION THREAD

**Discussion by lacell75** [webdriverio/webdriverio#6223] [Ideas]
**Improve the Shadow DOM support :**
The WebDriver protocol does not provide native support for shadow DOM and I think it's [abandoned](https://github.com/w3c/webdriver/pull/1320), so I would like and I think the `shadow$` and `shadow$$` commands can be improve.
Currently, we need to declare all shadowRoot tree to interact with an element's shadow DOM (Declared in pages/components objects) .  My proposable is to have a element's shadow DOM selection more "intelligent", able to browse the entire tree of shadowRoots in all DOM without declared all shadowRoots and able to find the element expected. (like the $ $$ commands). I think it's technically feasible and the development time savings would be huge for us. WDYT?

**Comment by christian-bromann** [comment]
@lacell75 yesterday a [patch to the spec](https://github.com/w3c/webdriver/pull/1565) has been merged that will enable support for shadow DOM. It doesn't seem that it allows your suggestion of browse all shadow roots and afaik this is also not possible. Can you elaborate how this can be implemented?

**Comment by lacell75** [comment]
Is it a dream or you are really implemented a new support for the shadow DOM in v7.5.0 release? Its incredible..
I'm very excited to test that.... What about performance to find the element?

**Reply by christian-bromann** [reply]
I don't think there should be a big performance burden given that it is a script running in the browser. Mind commenting on #6733

**Comment by christian-bromann** [comment]
With the implementation of https://webdriver.io/docs/selectors/#deep-selectors can this be marked as answered?

**Reply by lacell75** [reply]
sure
